packages/clerk-js/src/ui/elements/Switch.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { forwardRef, useState } from 'react';
+
+import type { LocalizationKey } from '../customizables';
+import { descriptors, Flex, Text } from '../customizables';
+
+interface SwitchProps {
+  checked?: boolean;
+  defaultChecked?: boolean;
+  onChange?: (checked: boolean) => void;
+  disabled?: boolean;
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+  title?: string | LocalizationKey;
+}
+
+export const Switch = forwardRef<HTMLDivElement, SwitchProps>(
+  (
+    {
+      checked: controlledChecked,
+      defaultChecked,
+      onChange,
+      disabled = false,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledby,
+      title,
+    },
+    ref,
+  ) => {
+    const [internalChecked, setInternalChecked] = useState(!!defaultChecked);
+    const isControlled = controlledChecked !== undefined;
+    const checked = isControlled ? controlledChecked : internalChecked;
+
+    const handleToggle = (_: React.MouseEvent | React.KeyboardEvent) => {
+      if (disabled) return;
+      if (!isControlled) {
+        setInternalChecked(!checked);
+      }
+      onChange?.(!checked);
+    };
+
+    const handleKeyDown = (e: React.KeyboardEvent) => {
+      if (e.key === ' ' || e.key === 'Enter') {
+        e.preventDefault();
+        handleToggle(e);
+      }
+    };
+
+    return (
+      <Flex
+        direction='row'
+        align='center'
+        as='div'
+        ref={ref}
+        tabIndex={disabled ? -1 : 0}
+        role='switch'
+        aria-checked={checked}
+        aria-label={ariaLabel}
+        aria-labelledby={ariaLabelledby}
+        elementDescriptor={descriptors.switchRoot}
+        onClick={handleToggle}
+        onKeyDown={handleKeyDown}
+        sx={{
+          width: 'fit-content',
+          userSelect: 'none',
+        }}
+      >
+        <Flex
+          as='div'
+          align='center'
+          sx={t => ({
+            width: 24,
+            height: 16,
+            alignItems: 'center',
+            position: 'relative',
+            backgroundColor: checked ? t.colors.$primary500 : t.colors.$neutralAlpha150,
```

Comment:
Folks will need a way to target the background color when in the checked state. There is some precedence for `applyDataStateProps` in some of the components that have states that allow users to target.